File: challenges/codeabbey/056/raguar94.cpp
#include <iostream>
using namespace std;

int main(){
  string board[5];
  for (int i=0;i<5;i++){
    cin >> board[i];
  }
  int turns = 5;
  char board2[50][50];
  for (int i = 0; i < 50; i++) {
    for (int j = 0; j < 50; j++) {
      if ((i >=23 && i <= 27) && (j >= 23 && j <= 29)){
        board2[i][j] = board[i-23][j-23];
      } else {
        board2[i][j] = '-';
        }
      }
    }
  int organisms;
  int grimm;
  for (int i=0;i<turns;i++){
    organisms = 0;
    for (int j=1;j<49;j++){
      for (int k=1;k<49;k++){
        grimm = 0;
        if (board2[j][k+1] == 'X' || board2[j][k+1] == 'D') {
          grimm += 1;
        }
        if (board2[j+1][k] == 'X' || board2[j+1][k] == 'D') {
          grimm += 1;
        }
        if (board2[j+1][k+1] == 'X' || board2[j+1][k+1] == 'D') {
          grimm += 1;
        }
        if (board2[j][k-1] == 'X' || board2[j][k-1] == 'D') {
          grimm += 1;
        }
        if (board2[j-1][k] == 'X' || board2[j-1][k] == 'D') {
          grimm += 1;
        }
        if (board2[j-1][k-1] == 'X' || board2[j-1][k-1] == 'D') {
          grimm += 1;
        }
        if (board2[j-1][k+1] == 'X' || board2[j-1][k+1] == 'D') {
          grimm += 1;
        }
        if (board2[j+1][k-1] == 'X' || board2[j+1][k-1] == 'D') {
          grimm += 1;
        }
        if ((grimm < 2 || grimm > 3) && (board2[j][k] == 'X')){
          board2[j][k] = 'D';
        }
        if (board2[j][k] == '-' && grimm == 3){
          board2[j][k] = 'B';
        }
      }
    }
    for (int i=1;i<49;i++){
      for (int j=1;j<49;j++){
        if (board2[i][j] == 'D'){
          board2[i][j] = '-';
        }
        if (board2[i][j] == 'B') {
          board2[i][j] = 'X';
        }
      }
    }
    for (int i=1;i<49;i++){
      for (int j=1;j<49;j++){
        if (board2[i][j] == 'X') {
          organisms += 1;
        }
      }
    }
    cout << organisms << " ";
  }
}

File: challenges/codeabbey/056/raguar94.py
"""This code shows the number of organisms remaining at the end of the turn
 for a certain quantity of turns, the remaining organisms are obtained
 by the rules of game of life by Jhon Conway"""
turn = 5
board = []
original = []
for i in range(5):
    origina.append(raw_input())
for i in range(50):
    for j in range(50):
        board.append([])
        if (i >= 23 and i <= 27) and (j >= 23 and j <= 29):
            board[i].append(origina[i-23][j-23])
        else:
            board[i].append("-")
for i in range(turn):
    organism = 0
    for j in range(1, 49):
        for k in range(1, 49):
            grimm = 0
            if (board[j][k+1] == "X") or (board[j][k+1] == "D"):
                grimm += 1
            if (board[j+1][k] == "X") or (board[j+1][k] == "D"):
                grimm += 1
            if (board[j+1][k+1] == "X") or (board[j+1][k+1] == "D"):
                grimm += 1
            if (board[j][k-1] == "X") or (board[j][k-1] == "D"):
                grimm += 1
            if (board[j-1][k] == "X") or (board[j-1][k] == "D"):
                grimm += 1
            if (board[j-1][k-1] == "X") or (board[j-1][k-1] == "D"):
                grimm += 1
            if (board[j-1][k+1] == "X") or (board[j-1][k+1] == "D"):
                grimm += 1
            if (board[j+1][k-1] == "X") or (board[j+1][k-1] == "D"):
                grimm += 1
            if (grimm < 2 or grimm > 3) and (board[j][k] == "X"):
                board[j][k] = "D"
            if (grimm == 3) and (board[j][k] == "-"):
                board[j][k] = "B"
    for j in range(1, 49):
        for k in range(1, 49):
            if board[j][k] == "D":
                board[j][k] = "-"
            if board[j][k] == "B":
                board[j][k] = "X"
    for j in range(50):
        for k in range(50):
            if board[j][k] == "X":
                organism += 1
    print str(organism) + " "
